### PR TITLE
Unify user session lifetime configuration

### PR DESCRIPTION
**Pull request by michalmisiewicz**
In current version of Airflow user session lifetime can be configured by `session_lifetime_days` and `force_log_out_after options`. In practise only `session_lifetime_days` has impact on session lifetime, but it is limited to values in day. In this PR I have simplified user session lifetime configuration.

### CODE REVIEW

**Comment by michalmisiewicz** [comment]
@mik-laj I have to recreate the PR

**Review by mik-laj — APPROVED** [review]

File: airflow/www/app.py
Comment (XD-DENG) [inline airflow/www/app.py]: Instead of printing a warning, would it be better to make it a "hard stopper"?

My concern is some people normally simply skip warning messages, in  which case, the system behaviour would be different from what they expect.

**Review by XD-DENG — CHANGES_REQUESTED** [review]

File: airflow/www/app.py
Comment (XD-DENG) [reply airflow/www/app.py]: e.g. say I'm still using `SESSION_LIFETIME_DAYS` in my configuration, with non-default value, say 2 days. Assuming I'm a person who's used to skipping warnings, then what would actually happen is the session lifetime is gonna be 43200 min, i.e. 30 days, not what I think I'm configuring (2 days).

If I missed anything, let me know please. Thanks.

**Review by tooptoop4 — APPROVED** [review]

**Comment by kaxil** [comment]
Can you please rebased your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Comment by michalmisiewicz** [comment]
@kaxil I've rebased with master

**Comment by mik-laj** [comment]
@michalmisiewicz One comment left. https://github.com/apache/airflow/pull/11970#discussion_r515365185 Can you reply, please?

**Comment by XD-DENG** [comment]
Hi @mik-laj , would like to have your opinion as well: if you find my comment above makes sense, let's wait for the contributor to reply; But if you find the current implementation is good enough, I'm ok to have it merged.

**Comment by mik-laj** [comment]
@XD-DENG Both approaches are fine for me. I have no preferences. I can see that @msumit  gave the thumbs up, so I guess it'll be better if we accept your proposal.

File: airflow/www/app.py
Comment (michalmisiewicz) [reply airflow/www/app.py]: @XD-DENG @mik-laj Sorry I was busy. I agree with @XD-DENG.  We should stop app when we detect removed options in configuration.

Comment (XD-DENG) [reply airflow/www/app.py]: Cool, we are aligned then.

Take your time please for the implementation if you are busy. Thanks!

Comment (michalmisiewicz) [reply airflow/www/app.py]: @XD-DENG @mik-laj I've updated the code

File: airflow/www/extensions/stop_webserver.py
Comment (XD-DENG) [inline airflow/www/extensions/stop_webserver.py]: Personally I don't think it's necessary to have such a new file/new function only for `exit with code 4`.

**Review by XD-DENG — CHANGES_REQUESTED** [review]

File: airflow/www/app.py
Comment (XD-DENG) [inline airflow/www/app.py]: A minor one: I suggest to make the name more explicit, say `default_session_lifetime_mins`.

File: airflow/www/app.py
Comment (XD-DENG) [inline airflow/www/app.py]: This `if-else` condition check is allowing one situation which should not happen in my opinion: if a user has both `SESSION_LIFETIME_MINUTES` AND `SESSION_LIFETIME_DAYS`/`FORCE_LOG_OUT_AFTER` in his/her configuration, NO error will be raised. Of course it will correctly use `SESSION_LIFETIME_MINUTES`. However, when another user looks at this configuration, it will be confusing and misleading.

So I would like to suggest change the `if-else` check into structure below:

```python
if conf has SESSION_LIFETIME_DAYS or FORCE_LOG_OUT_AFTER:
    logging.error() + exit()

if conf.has_option('webserver', 'SESSION_LIFETIME_MINUTES'):
    session_lifetime_minutes = conf.getint('webserver', 'SESSION_LIFETIME_MINUTES', ......)
else:
    logging.info("Using default value for SESSION_LIFETIME_MINUTES)
    session_lifetime_minutes = default_session_lifetime
```

Let me know your thoughts? Thanks.

**Review by XD-DENG — CHANGES_REQUESTED** [review]

File: airflow/www/extensions/stop_webserver.py
Comment (michalmisiewicz) [reply airflow/www/extensions/stop_webserver.py]: I wanted to hide `sys.exit(4)` to function because without gunicorn documentation 4 is a magic number. I can inline it and pass link to url to sourcecode in comment.

File: airflow/www/app.py
Comment (michalmisiewicz) [reply airflow/www/app.py]: This is how it looked in the first version of PR. 😉 Later with @mik-laj we came to [conclusion ](https://github.com/apache/airflow/pull/11745#discussion_r510427359) that we should not penalize users when `SESSION_LIFETIME_MINUTES` is passed. It's hard to say which approach is better.  
After all I think it will be better to stop execution when removed options are provided because it will enforce users to update confiuration and it will be less error pruning.

Comment (XD-DENG) [reply airflow/www/app.py]: Maybe @mik-laj shares if you have different opinion please.

Comment (michalmisiewicz) [reply airflow/www/app.py]: @mik-laj Could you share your thoughts ?

Comment (mik-laj) [reply airflow/www/app.py]: I agree. We can raise an exception when the configuration contains options that we no longer support.

Comment (michalmisiewicz) [reply airflow/www/app.py]: @XD-DENG I've just updated the code.

File: UPDATING.md
Comment (XD-DENG) [inline UPDATING.md]: Hi @michalmisiewicz . Thanks for the update. 

All look good to me other than a last point: this change update should be put under section `## Airflow Master` of `UPDATING.md`, rather than `## Airflow 2.0.0a1`. The release manager(s) will decide when and which version to include this.

May you please make this change? Please also remember to rebase to the latest Master branch. After that, I think we are good to merge.

Thanks!

**Review by XD-DENG — CHANGES_REQUESTED** [review]

File: UPDATING.md
Comment (XD-DENG) [reply UPDATING.md]: Please also remove "In Airflow 2.0", given in which version we should include this change is not determined yet. Thanks.

**Review by XD-DENG — APPROVED** [review]
Many thanks @michalmisiewicz

File: airflow/www/app.py
Comment (ashb) [inline airflow/www/app.py:84]: NOOOOOOOOO. Upgrade, don't error.

**Pull request merged**
